src/crimson/os/seastore/cache.cc

```diff
@@ -1709,12 +1711,15 @@ void Cache::init()
     remove_extent(root);
     root = nullptr;
   }
+  end_to_end_data_protection =
+    epm.get_primary_device().is_end_to_end_data_protection();
```

Comment:
Seastore is supposed to use multiple tiers of devices (there can be different types of devices at the same time), so I think the design should support per-device decisions about whether the crc should be calculated/validated or not.